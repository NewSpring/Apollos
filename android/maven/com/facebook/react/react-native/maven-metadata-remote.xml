<?xml version="1.0" encoding="UTF-8"?>
<metadata>
  <groupId>com.facebook.react</groupId>
  <artifactId>react-native</artifactId>
  <versioning>
    <release>28.0.0</release>
    <versions>
      <version>28.0.0</version>
    </versions>
    <lastUpdated>20180702224301</lastUpdated>
  </versioning>
</metadata>
