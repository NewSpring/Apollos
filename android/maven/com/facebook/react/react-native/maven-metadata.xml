<?xml version="1.0" encoding="UTF-8"?>
<metadata>
  <groupId>com.facebook.react</groupId>
  <artifactId>react-native</artifactId>
  <versioning>
    <release>25.0.0</release>
    <versions>
      <version>25.0.0</version>
    </versions>
    <lastUpdated>20180118183717</lastUpdated>
  </versioning>
</metadata>
