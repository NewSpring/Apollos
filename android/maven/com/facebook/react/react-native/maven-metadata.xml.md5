4d33011c5cd3744e3f0a6d3462b247db