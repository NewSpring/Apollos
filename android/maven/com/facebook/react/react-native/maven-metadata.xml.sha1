8ec2ecca6c881de314f48a8ab786d36bf4a50ada