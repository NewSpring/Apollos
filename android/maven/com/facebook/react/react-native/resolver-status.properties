#NOTE: This is an internal implementation file, its format can be changed without prior notice.
#Thu Jan 18 10:37:17 PST 2018
maven-metadata-remote.xml.lastUpdated=1516300637918
