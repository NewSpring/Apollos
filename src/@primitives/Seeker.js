import React, { Component } from 'react';
import {
  View,
  PanResponder,
} from 'react-native';
import PropTypes from 'prop-types';
import withTheme from './withTheme';

export class Seeker extends Component {
  static propTypes = {
    progress: PropTypes.number,
    onSeek: PropTypes.func,
    trackHeight: PropTypes.number,
    trackColor: PropTypes.string,
    progressHeight: PropTypes.number,
    progressColor: PropTypes.string,
    knobColor: PropTypes.string,
    knobRadius: PropTypes.number,
    knobSize: PropTypes.number,
  };

  static defaultProps = {
    progress: 0,
    onSeek() {},
    trackHeight: 20,
    trackColor: 'gray',
    progressHeight: 20,
    progressColor: 'red',
    knobColor: 'blue',
    knobRadius: 9999,
    knobSize: 30,
  };

  state = {
    position: 0,
    offset: 0,
    isSeeking: false,
    width: 0,
  }

  componentWillReceiveProps({ progress }) {
    if (progress !== this.props.progress) {
      this.setState({
        position: progress * this.state.width,
      });
    }
  }

  get currentProgress() {
    return this.state.position / this.state.width;
  }

  panResponder = PanResponder.create({
    onStartShouldSetPanResponder: () => (true),
    onPanResponderMove: (e, { dx }) => {
      this.setState({
        offset: dx,
        isSeeking: true,
      });
    },
    onPanResponderRelease: (e, { dx }) => {
      this.setState({
        position: this.state.position + dx,
        offset: 0,
        isSeeking: false,
      }, () => {
        const { onSeek } = this.props;
        if (onSeek) onSeek(this.currentProgress);
      });
    },
  });

  handleOnLayout = ({ nativeEvent: { layout: { width } } }) => {
    this.setState({
      position: ((this.state.position / this.state.width) * width || this.props.progress * width),
      width,
    });
  };

  render() {
    const {
      trackHeight,
      trackColor,
      progressHeight,
      progressColor,
      knobColor,
      knobRadius,
      knobSize,
    } = this.props;

    const {
      position,
      offset,
    } = this.state;

    return (
      <View onLayout={this.handleOnLayout}>
        <View
          style={{
            backgroundColor: trackColor,
            height: trackHeight,
          }}
        />
        <View
          style={{
            position: 'absolute',
            left: 0,
            backgroundColor: progressColor,
            height: progressHeight,
            width: position + offset,
            transform: [
              { translateY: (trackHeight - progressHeight) / 2 },
            ],
          }}
        />
        <View
          style={{
            backgroundColor: knobColor,
            borderRadius: knobRadius,
            position: 'absolute',
            left: (position - (knobSize / 2)) + offset,
            height: knobSize,
            width: knobSize,
            transform: [
              { translateY: (trackHeight - knobSize) / 2 },
            ],
          }}
          {...this.panResponder.panHandlers}
        />
      </View>
    );
  }
}

export default withTheme(({ theme, ...otherProps } = {}) => ({
  progressColor: theme.primaryColor,
  knobColor: theme.secondaryColor,
  trackColor: theme.darkPrimaryColor,
  ...otherProps,
}))(Seeker);
