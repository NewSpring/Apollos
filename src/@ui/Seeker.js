import React, { Component } from 'react';
import {
  View,
  PanResponder,
} from 'react-native';
import PropTypes from 'prop-types';
import { withTheme } from '@ui/theme';

export class Seeker extends Component {
  static propTypes = {
    progress: PropTypes.number,
    onSeek: PropTypes.func,
    onSeeking: PropTypes.func,
    trackHeight: PropTypes.number,
    trackColor: PropTypes.string,
    progressHeight: PropTypes.number,
    progressColor: PropTypes.string,
    knobColor: PropTypes.string,
    knobRadius: PropTypes.number,
    knobSize: PropTypes.number,
  };

  static defaultProps = {
    progress: 0,
    onSeek() {},
    onSeeking() {},
    trackHeight: 20,
    trackColor: 'gray',
    progressHeight: 20,
    progressColor: 'red',
    knobColor: 'blue',
    knobRadius: 9999,
    knobSize: 30,
  };

  state = {
    position: 0,
    offset: 0,
    width: 0,
  }

  componentWillReceiveProps({ progress }) {
    if (progress !== this.props.progress) {
      this.setState({
        position: progress * this.state.width,
      });
    }
  }

  get currentProgress() {
    return this.state.position / this.state.width;
  }

  panResponder = PanResponder.create({
    onStartShouldSetPanResponder: () => (true),
    onPanResponderMove: (e, { dx }) => {
      this.setState({
        offset: dx,
      });
      const { onSeeking } = this.props;
      onSeeking((this.state.position + dx) / this.state.width);
    },
    onPanResponderRelease: (e, { dx }) => {
      this.setState({
        position: this.state.position + dx,
        offset: 0,
      }, () => {
        const { onSeek } = this.props;
        if (onSeek) onSeek(this.currentProgress);
      });
    },
  });

  handleOnLayout = ({ nativeEvent: { layout: { width } } }) => {
    this.setState({
      position: ((this.state.position / this.state.width) * width || this.props.progress * width),
      width,
    });
  };

  render() {
    const {
      trackHeight,
      trackColor,
      progressHeight,
      progressColor,
      knobColor,
      knobRadius,
      knobSize,
    } = this.props;

    const {
      position,
      offset,
    } = this.state;

    return (
      <View onLayout={this.handleOnLayout}>
        <View
          style={{
            backgroundColor: trackColor,
            height: trackHeight,
          }}
        />
        <View
          style={{
            position: 'absolute',
            left: 0,
            backgroundColor: progressColor,
            height: progressHeight,
            width: position + offset,
            transform: [
              { translateY: (trackHeight - progressHeight) / 2 },
            ],
          }}
        />
        <View
          style={{
            backgroundColor: knobColor,
            borderRadius: knobRadius,
            position: 'absolute',
            left: (position - (knobSize / 2)) + offset,
            height: knobSize,
            width: knobSize,
            transform: [
              { translateY: (trackHeight - knobSize) / 2 },
            ],
          }}
          {...this.panResponder.panHandlers}
        />
      </View>
    );
  }
}

export default withTheme(({ theme, ...otherProps } = {}) => ({
  progressColor: theme.colors.common.primary,
  knobColor: theme.colors.common.secondary,
  trackColor: theme.colors.common.darkPrimary,
  ...otherProps,
}))(Seeker);
