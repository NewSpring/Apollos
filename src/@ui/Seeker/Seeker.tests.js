import React from 'react';
import renderer from 'react-test-renderer';
import { Animated } from 'react-native';
import Providers from '@ui/TestProviders';

import Seeker from './';

describe('The Seeker component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <Seeker
          progress={new Animated.Value(0)}
        />
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
});
