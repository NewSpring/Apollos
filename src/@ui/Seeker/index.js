import React, { PureComponent } from 'react';
import {
  Animated,
  View,
  PanResponder,
  StyleSheet,
} from 'react-native';
import PropTypes from 'prop-types';
import Color from 'color';
import styled from '@ui/styled';
import { withTheme } from '@ui/theme';

const Container = styled(({ theme }) => ({
  minWidth: '65%',
  marginHorizontal: theme.sizing.baseUnit,
  overflow: 'visible',
  paddingHorizontal: theme.sizing.baseUnit / 2,
  paddingVertical: (theme.sizing.baseUnit / 2) - (theme.sizing.borderRadius / 2),
}), 'Seeker.Container')(View);

const Track = styled(({ theme }) => ({
  backgroundColor: Color(theme.colors.text.primary).fade(theme.alpha.medium).string(),
  height: theme.sizing.borderRadius,
  borderRadius: theme.sizing.borderRadius,
  overflow: 'hidden',
}), 'Seeker.Track')(View);

const ProgressBar = styled(({ theme }) => ({
  backgroundColor: theme.colors.text.primary,
  ...StyleSheet.absoluteFillObject,
}), 'Seeker.ProgressBar')(View);

const Knob = styled(({ theme }) => ({
  backgroundColor: theme.colors.text.primary,
  borderRadius: theme.sizing.baseUnit,
  position: 'absolute',
  top: 0,
  right: 0,
  height: theme.sizing.baseUnit,
  width: theme.sizing.baseUnit,
  elevation: 2,
  zIndex: 100,
}), 'Seeker.Knob')(View);

export class Seeker extends PureComponent {
  static propTypes = {
    progress: PropTypes.object, // eslint-disable-line
    onSeek: PropTypes.func,
    onSeeking: PropTypes.func,
  };

  static defaultProps = {
    progress: new Animated.Value(0),
    onSeek() {},
    onSeeking() {},
  };

  state = {
    width: 0,
  }

  componentWillMount() {
    this.listen(this.props.progress);
  }

  componentWillReceiveProps({ progress }) {
    if (progress !== this.props.progress) {
      this.listen(progress);
    }
  }

  componentWillUnmount() {
    if (this.listener) this.props.progress.removeListener(this.props.progress);
  }

  get trackBarOffset() {
    const progressInvert = Animated.add(1, Animated.multiply(this.props.progress, -1));
    const position = Animated.multiply(progressInvert, -this.state.width);
    return Animated.add(position, this.offsetDriver);
  }

  get knobStyles() {
    return ({
      position: 'absolute',
      right: 0,
      top: 0,
      bottom: 0,
      width: '100%',
      overflow: 'visible',
      transform: [
        { translateX: this.trackBarOffset },
      ],
    });
  }

  get progressBarStyles() {
    return ([StyleSheet.absoluteFill, {
      transform: [{ translateX: this.trackBarOffset }],
    }]);
  }

  listen = (progress) => {
    if (this.listener) this.props.progress.removeListener(progress);
    this.listener = progress.addListener(({ value }) => {
      this.lastProgressValue = value;
      this.lastPosition = value * this.state.width;
    });
  }

  offsetDriver = new Animated.Value(0);

  panResponder = PanResponder.create({
    onStartShouldSetPanResponder: () => !!this.props.onSeeking,
    onPanResponderMove: (e, { dx }) => {
      let offset = dx;
      offset = Math.min(this.state.width - this.lastPosition, offset);
      offset = Math.max(-this.lastPosition, offset);
      this.offsetDriver.setValue(offset || 0);

      const { onSeeking } = this.props;
      onSeeking((this.lastPosition + dx) / this.state.width);
    },
    onPanResponderRelease: (e, { dx }) => {
      this.offsetDriver.setValue(0);
      const { onSeek } = this.props;
      if (onSeek) onSeek(((this.lastPosition || 0) + dx) / this.state.width);
    },
  });

  handleOnLayout = ({ nativeEvent: { layout: { width } } }) => {
    this.setState({
      width,
    });
  };

  render() {
    return (
      <Container>
        <Track onLayout={this.handleOnLayout}>
          <Animated.View
            style={this.progressBarStyles}
          >
            <ProgressBar />
          </Animated.View>
        </Track>
        <Animated.View
          style={this.knobStyles}
        >
          <Knob {...this.panResponder.panHandlers} />
        </Animated.View>
      </Container>
    );
  }
}

export default withTheme(({ theme, ...otherProps } = {}) => ({
  progressColor: theme.colors.primary,
  knobColor: theme.colors.secondary,
  trackColor: theme.colors.darkPrimary,
  ...otherProps,
}))(Seeker);
