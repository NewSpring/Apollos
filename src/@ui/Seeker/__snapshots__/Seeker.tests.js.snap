// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Seeker component should render 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 20,
      "minWidth": "65%",
      "overflow": "visible",
      "paddingHorizontal": 10,
      "paddingVertical": 7,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(48, 48, 48, 0.30000000000000004)",
        "borderRadius": 6,
        "height": 6,
        "overflow": "hidden",
      }
    }
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#303030",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
  <View
    collapsable={undefined}
    style={
      Object {
        "bottom": 0,
        "overflow": "visible",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
        "width": "100%",
      }
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "backgroundColor": "#303030",
          "borderRadius": 20,
          "elevation": 2,
          "height": 20,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 20,
          "zIndex": 100,
        }
      }
    />
  </View>
</View>
`;
